src/Dapr.AspNetCore/DaprServiceCollectionExtensions.cs

```diff
@@ -11,55 +11,85 @@
-        /// <param name="configure"></param>
-        public static void AddDaprClient(this IServiceCollection services, Action<DaprClientBuilder> configure = null)
+        ArgumentNullException.ThrowIfNull(services, nameof(services));
+
+        services.TryAddSingleton(serviceProvider =>
         {
-            ArgumentNullException.ThrowIfNull(services, nameof(services));
-
-            services.TryAddSingleton(_ =>
-            {
-                var builder = new DaprClientBuilder();
-                configure?.Invoke(builder);
-
-                return builder.Build();
-            });
-        }
-        
-        /// <summary>
-        /// Adds Dapr client services to the provided <see cref="IServiceCollection"/>. This does not include integration
-        /// with ASP.NET Core MVC. Use the <c>AddDapr()</c> extension method on <c>IMvcBuilder</c> to register MVC integration. 
-        /// </summary>
-        /// <param name="services">The <see cref="IServiceCollection"/>.</param>
-        /// <param name="configure"></param>
-        public static void AddDaprClient(this IServiceCollection services,
-            Action<IServiceProvider, DaprClientBuilder> configure)
+            var builder = new DaprClientBuilder();
+
+            var configuration = serviceProvider.GetService<IConfiguration>();
+
+            //Set the HTTP endpoint, if provided, else use the default endpoint
+            builder.UseHttpEndpoint(DaprDefaults.GetDefaultHttpEndpoint(configuration));
+
+            //Set the gRPC endpoint, if provided
+            builder.UseGrpcEndpoint(DaprDefaults.GetDefaultGrpcEndpoint(configuration));
+            
+            //Set the API token, if provided
+            var apiToken = DaprDefaults.GetDefaultDaprApiToken(configuration);
+            if (!string.IsNullOrWhiteSpace(apiToken))
+                builder.UseDaprApiToken(apiToken);
+
+            configure?.Invoke(builder);
+
+            return builder.Build();
+        });
+    }
+
+    /// <summary>
+    /// Adds Dapr client services to the provided <see cref="IServiceCollection"/>. This does not include integration
+    /// with ASP.NET Core MVC. Use the <c>AddDapr()</c> extension method on <c>IMvcBuilder</c> to register MVC integration. 
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/>.</param>
+    /// <param name="configure"></param>
+    public static void AddDaprClient(this IServiceCollection services,
+        Action<IServiceProvider, DaprClientBuilder> configure)
+    {
+        ArgumentNullException.ThrowIfNull(services, nameof(services));
+
+        services.TryAddSingleton(serviceProvider =>
         {
-            ArgumentNullException.ThrowIfNull(services, nameof(services));
+            var builder = new DaprClientBuilder();
+
+            var configuration = serviceProvider.GetService<IConfiguration>();
+
+            //Set the HTTP endpoint, if provided, else use the default endpoint
```

Comment:
Nit: the actual configuration logic is the same in both `AddDaprClient()` overrides.  Consider consolidating into a single function?  (Similarly, in the Actors case.)

Reply:
Updated - I looked for an opportunity in the Workflow client as well, but it's got multiple builder interfaces so my consolidated approach wound up being about the same length.